src/lib/discovery.js

```diff
@@ -11,78 +11,159 @@ import EventEmitter from 'eventemitter3'
 import Client from './client'
 
 export class SKServer {
-  constructor (service) {
+  constructor(service) {
     this._roles = service.roles || ['master', 'main']
     this._self = service.self || ''
     this._version = service.version || '0.0.0'
     this._hostname = service.hostname
     this._port = service.port
   }
 
-  get roles () {
+  get roles() {
     return this._roles
   }
 
-  get self () {
+  get self() {
     return this._self
   }
 
-  get version () {
+  get version() {
     return this._version
   }
 
-  get hostname () {
+  get hostname() {
     return this._hostname
   }
 
-  get port () {
+  get port() {
     return this._port
   }
 
-  isMain () {
+  isMain() {
     return this._roles.includes('main')
   }
 
-  isMaster () {
+  isMaster() {
     return this._roles.includes('master')
   }
 
-  createClient (opts = {}) {
+  createClient(opts = {}) {
     return new Client({
       ...opts,
       hostname: this._hostname,
-      port: this._port
+      port: this._port,
     })
   }
 }
 
 export default class Discovery extends EventEmitter {
-  constructor (bonjour, timeout = 60000) {
+  constructor(bonjourOrMdns, timeout = 60000) {
     super()
 
-    const props = [ '_server', '_registry' ].join(',')
+    this.found = []
 
-    if (!bonjour || typeof bonjour !== 'object' || Object.keys(bonjour).join(',') !== props) {
-      throw new Error('Invalid mDNS provider')
+    if (!bonjourOrMdns || typeof bonjourOrMdns !== 'object') {
+      throw new Error('No mDNS provider given')
     }
 
-    this.found = []
+    const bonjourProps = ['_server', '_registry'].join(',')
+    const mdnsProps = ['dns_sd', 'Advertisement', 'createAdvertisement', 'Browser'].join(',')
+
```

Comment:
Why are we testing that the order of the keys is exactly this and these are the first keys?

Reply:
First keys was just bcs I didn't want to compare every key, seemed uncessary. My thinking re the order was that, if the order didn't match, chances are that it's a different module with similar methods or a very different version of the module. That way we have a loose version check, which would catch any major changes to those modules (i.e. fingerprinting)

I would have preferred a way to check the modules identity, but I don't think that's possible besides fingerprinting. If you know of a better way that allows us to check that a module is `mdns` or `bonjour`, and that the version is within a supported range, I'm all ears :)

Reply:
I would just call `hasOwnProperty` on the prop names and be done with it. Relying on the order seems needlessly brittle.